eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -136,31 +164,42 @@ conwayDelegTransition = do
       checkStakeKeyNotAlreadyRegistered stakeCred dsUnified
       pure $ dState {dsUnified = acceptDepositForStakeKey stakeCred dsUnified ppKeyDeposit}
     ConwayUnRegCert stakeCred sMayDeposit -> do
-      checkStakeKeyIsAlreadyRegistered stakeCred dsUnified
-      checkStakeKeyHasZeroBalance stakeCred dsUnified
+      checkStakeKeyIsRegistered stakeCred dsUnified
+      checkStakeKeyHasZeroBalanceReward stakeCred dsUnified
       forM_ sMayDeposit $ checkDepositAgainstPaidDeposit stakeCred dsUnified
       pure $
         dState
-          { dsUnified = UM.domDeleteAll (Set.singleton stakeCred) dsUnified
+          { dsUnified = UM.domDeleteAll (Set.singleton stakeCred) dsUnified -- QUESTION: @WhatisRT: We auto-un-register him as a DRep?
```

Comment:
This is stake credential deregistration, correct? If so, nothing should happen with the DRep registry.

Reply:
Oh got it! Sorry for the wrong question here. "unregistering" will result in not only removing the stake deposit and delegation, but also the DRep delegation. That's what I meant to ask about. `ConwayUnRegCert` currently removes even DRep delegation, because DRep delegation is also efficiently kept in the same UMap, that is keyed by stake credential.